src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/FindPal.cs

```diff
@@ -0,0 +1,405 @@
+                                CERT_POLICY_INFO* pCertPolicyInfo = &(pCertPoliciesInfo->rgPolicyInfo[i]);
+                                string actual = Marshal.PtrToStringAnsi(pCertPolicyInfo->pszPolicyIdentifier);
+                                if (oidValue.Equals(actual, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    foundMatch = true;
+                                    break;
+                                }
+                            }
+                        }
+                        ))
+                    {
+                        return false;
+                    }
+
+                    GC.KeepAlive(pCertContext);
+                    return foundMatch;
+                });
+        }
+
+        public unsafe void FindByExtension(string oidValue)
+        {
+            FindCore(
+                delegate (SafeCertContextHandle pCertContext)
+                {
+                    CERT_INFO* pCertInfo = pCertContext.CertContext->pCertInfo;
+                    CERT_EXTENSION* pCertExtension = Interop.crypt32.CertFindExtension(oidValue, pCertInfo->cExtension, pCertInfo->rgExtension);
+                    GC.KeepAlive(pCertContext);
+                    return pCertExtension != null;
+                });
+        }
+
+        public unsafe void FindByKeyUsage(X509KeyUsageFlags keyUsage)
+        {
+            FindCore(
+                delegate (SafeCertContextHandle pCertContext)
+                {
+                    CERT_INFO* pCertInfo = pCertContext.CertContext->pCertInfo;
+                    X509KeyUsageFlags actual;
+                    if (!Interop.crypt32.CertGetIntendedKeyUsage(CertEncodingType.All, pCertInfo, out actual, sizeof(X509KeyUsageFlags)))
+                        return true;  // no key usage means it is valid for all key usages.
+                    GC.KeepAlive(pCertContext);
+                    return (actual & keyUsage) == keyUsage;
+                });
+        }
+
+        public void FindBySubjectKeyIdentifier(byte[] keyIdentifier)
+        {
+            FindCore(
+                delegate (SafeCertContextHandle pCertContext)
+                {
+                    int cbData = 0;
+                    if (!Interop.crypt32.CertGetCertificateContextProperty(pCertContext, CertContextPropId.CERT_KEY_IDENTIFIER_PROP_ID, null, ref cbData))
+                        return false;
+
+                    byte[] actual = new byte[cbData];
+                    if (!Interop.crypt32.CertGetCertificateContextProperty(pCertContext, CertContextPropId.CERT_KEY_IDENTIFIER_PROP_ID, actual, ref cbData))
+                        return false;
+
+                    return keyIdentifier.ContentsEqual(actual);
+                });
+        }
+
+        public void Dispose()
+        {
+            _storePal.Dispose();
+        }
+
+        private void FindCore(Func<SafeCertContextHandle, bool> filter)
+        {
+            unsafe
```

Comment:
> You have [unsafe] in the method signature in other FindCore methods... why not here, too?

The other methods take a void* parameter, so they require it.  By not putting `unsafe` in the declaration here it allows the callers using this one to not also need to say unsafe. 

At least, that's the motivation I assume, since I just moved the existing code to a new file :)